org.eclipse.ui.console/plugin.xml

```diff
@@ -92,6 +118,87 @@ M4 = Platform-specific fourth key
-		<initializer class="org.eclipse.ui.internal.console.ConsoleUIPreferenceInitializer"/>
-	</extension>
+      <initializer class="org.eclipse.ui.internal.console.ConsoleUIPreferenceInitializer"/>
+      <initializer class="org.eclipse.ui.internal.console.ansi.preferences.AnsiConsolePreferenceInitializer" />
+    </extension>
+
+    <extension
+        point="org.eclipse.ui.menus">
+      <menuContribution
+          allPopups="false"
+          locationURI="toolbar:org.eclipse.ui.console.ConsoleView">
+        <command
+            commandId="AnsiConsole.command.copy_with_escapes"
+            icon="icons/ansiconsolecopy.png"
+            style="push">
+        </command>
+        <command
+            commandId="AnsiConsole.command.enable_disable"
+            icon="icons/ansiconsole.png"
+            style="toggle">
+        </command>
+        <command
+            commandId="AnsiConsole.command.show_quick_settings"
+            icon="icons/ansiconsoleprefs.png"
+            style="push">
+        </command>
+      </menuContribution>
+    </extension>
+
+    <extension
+        point="org.eclipse.ui.commands">
+      <category
+           id="AnsiConsole.command.categoryid"
+           name="%command.category.name">
+      </category>
+      <command
+          categoryId="AnsiConsole.command.categoryid"
+          defaultHandler="org.eclipse.ui.internal.console.ansi.commands.CopyWithEscapesHandler"
+          description="%command.copy_with_escapes.description"
+          id="AnsiConsole.command.copy_with_escapes"
+          name="%command.copy_with_escapes.name">
+      </command>
+      <command
+          categoryId="AnsiConsole.command.categoryid"
+          defaultHandler="org.eclipse.ui.internal.console.ansi.commands.CopyWithoutEscapesHandler"
+          description="%command.copy_without_escapes.description"
+          id="AnsiConsole.command.copy_without_escapes"
+          name="%command.copy_without_escapes.name">
+      </command>
+      <command
+          categoryId="AnsiConsole.command.categoryid"
+          defaultHandler="org.eclipse.ui.internal.console.ansi.commands.EnableDisableHandler"
+          description="%command.enable_disable.description"
+          id="AnsiConsole.command.enable_disable"
+          name="%command.enable_disable.name">
+        <state
+            class="org.eclipse.ui.handlers.RegistryToggleState:true"
+            id="org.eclipse.ui.commands.toggleState">
+        </state>
+      </command>
+      <command
+          categoryId="AnsiConsole.command.categoryid"
+          defaultHandler="org.eclipse.ui.internal.console.ansi.commands.ShowQuickPrefsHandler"
+          description="%command.show_quick_settings.description"
+          id="AnsiConsole.command.show_quick_settings"
+          name="%command.show_quick_settings.name">
+      </command>
+    </extension>
+
+    <extension point="org.eclipse.ui.preferencePages">
+      <page name="%ansicon.prefpage.name"
```

Comment:
Can you please set a parent category so it becomes a child of the existing console preference page?

Reply:
Done.

Reply:
I'm not sure what the rules are for Eclipse.
In other projects I've worked on the one who opened the comment / issue gets to accept it and resolve it.
So I marked all entries that I've solved with "Done", and you resolve it.
If the rule is that I resolve, then I'll do that next round.